sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubCoderProviderRegistrar.java

```diff
@@ -37,6 +37,7 @@ public List<CoderProvider> getCoderProviders() {
             TypeDescriptor.of(PubsubMessage.class), PubsubMessageWithMessageIdCoder.of()),
         CoderProviders.forCoder(
             TypeDescriptor.of(PubsubMessage.class),
-            PubsubMessageWithAttributesAndMessageIdCoder.of()));
+            PubsubMessageWithAttributesAndMessageIdCoder.of()),
+        CoderProviders.forCoder(TypeDescriptor.of(PubsubMessage.class), PubsubMessageCoder.of()));
```

Comment:
moved this code to be the last in the registrar (rather than first), otherwise there are failures in the tests.  @johnjcasey Do you have any further info on coder registrars and how a coder is resolved when multiple coders are registered for the same class?

Reply:
@TheNeuralBit I think this is the area where there is lack of clarity.  As best as I can tell, here there are multiple coders registered for the same class (`PubsubMessage.class`).  It seems like order matters here, but I'm not sure all the implications.  Ex. if I put the new coder first in the registrar, existing pubsub unit tests will fail with coder errors.

Reply:
Thanks for the info @TheNeuralBit , this is super helpful context.  Sounds like what's here then is not going to break anything, which is the most important.

One aspect that I predict will be confusing for users is that if one attempts to _write_ pubsub messages with ordering key, the `PubsubMessageWithAttributesCoder` will be used by default unless `setCoder` is called; in that default case, the orderingKey will be "mysteriously" stripped away by the coder.  I suppose we might also need a new `write` helper method like `writeMessagesWithOrderingKeys` in addition to `writeMessages` in order help users to set the correct coder?

Reply:
Filed https://github.com/apache/beam/issues/23525 for the follow-up